Use "enum class" instead of plain enum for ThemeTypes

mdubet opened the pull request:
#### fea0e1d56ed57b95191a9a593e0cd7a6e8b2c975
<pre>
Use &quot;enum class&quot; instead of plain enum for ThemeTypes
<a href="https://bugs.webkit.org/show_bug.cgi?id=257784">https://bugs.webkit.org/show_bug.cgi?id=257784</a>
rdar://110378475

Reviewed by Youenn Fablet.

* Source/WebCore/platform/ThemeTypes.cpp:
(WebCore::operator&lt;&lt;):
* Source/WebCore/platform/ThemeTypes.h

Canonical link: <a href="https://commits.webkit.org/264939@main">https://commits.webkit.org/264939@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/af2befb3e4f3ea4a63013b1dc9ab1f14fa287704

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/9194 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/9476 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/9704 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/10852 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/9111 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/9203 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/11465 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/9444 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/11977 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/9342 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/10291 "Passed tests") | [⏳ 🧪 api-mac](https://ews-build.webkit.org/#/builders/API-Tests-macOS-EWS "Waiting to run tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/11008 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/7587 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/8389 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/15855 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/8687 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/8540 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/11862 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/9039 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/7373 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/8265 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2231 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/12489 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/8797 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk commented on Source/WebCore/platform/ThemeTypes.h:
```diff
@@ -33,12 +33,12 @@ class TextStream;
-enum SelectionPart {
+enum class SelectionPart : bool {
     SelectionBackground,
     SelectionForeground
```
Since they are now namespaced it would be better to rename these to Background and Foreground, no? Same below.

mdubet replied:
I don't have a real opinion on this 😐

mdubet replied:
Renamed ✅

vitorroriz commented on Source/WebCore/platform/ThemeTypes.cpp:
```diff
@@ -33,67 +33,69 @@ namespace WebCore {
 TextStream& operator<<(TextStream& ts, ThemeColor themeColor)
 {
+    // Use "using enum" when supported by GCC 11
+    // https://en.cppreference.com/w/cpp/language/enum#Using-enum-declaration
```
I think this comment is not necessary unless you transform it in a FIXME with an associated radar and/or bugzilla for tracking it. Otherwise, there is a good chance we will just forget about it.

mdubet replied:
I've removed it. It will apply to every enum class in the future anyway so it's not really pertinent to have it here.

vitorroriz commented on Source/WebCore/platform/ThemeTypes.h:
```diff
@@ -33,12 +33,12 @@ class TextStream;
 };
 
-enum ThemeFont {
+enum class ThemeFont {
```
```suggestion
enum class ThemeFont : uint8_t {
```

mdubet replied:
Fixed, and also for `ThemeColor`

youennf reviewed (approved):
We could think of moving from ThemeFont::CaptionFont to the shorter ThemeFont::Caption if that reads equally fine. Ditto for ThemeColor.

mdubet committed:
Use "enum class" instead of plain enum for ThemeTypes
https://bugs.webkit.org/show_bug.cgi?id=257784
rdar://110378475

Reviewed by Youenn Fablet.

* Source/WebCore/platform/ThemeTypes.cpp:
(WebCore::operator<<):
* Source/WebCore/platform/ThemeTypes.h

Canonical link: https://commits.webkit.org/264939@main
